File: TicketCatalogueService/src/main/kotlin/it/polito/wa2/lab5/group09/ticketcatalogueservice/entities/TicketCatalogue.kt
package it.polito.wa2.lab5.group09.ticketcatalogueservice.entities

import org.bson.types.ObjectId
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.index.Indexed
import org.springframework.data.mongodb.core.mapping.Document


@Document(collection = "ticket_catalogue")
data class TicketCatalogue (
    @Id
    @Indexed
    val ticketId: ObjectId = ObjectId.get(),
    var type: String,
    var price: Double,
    var zones : String,
    val maxAge : Int? = null,
    val minAge : Int? = null,
)

File: client/courseprojectg09/src/model/TicketCatalogue.js
class TicketCatalogue {
    constructor(id, type, price, zones, maxAge, minAge) {
        this.id = id;
        this.type = type;
        this.price = price;
        this.zones = zones;
        this.maxAge = maxAge;
        this.minAge = minAge;
    }

    static from(json) {
        const ticketCatalogue = new TicketCatalogue();
        Object.assign(ticketCatalogue, json);
        return ticketCatalogue;
    }
}

export default TicketCatalogue;